### ISSUE TITLE
What is the training loss value of deeplsd training to the best state?

### ISSUE THREAD

**Issue by AIQuantScience** [cvg/DeepLSD#24]
When training the wireframe dataset, the learning rate drops slowly under the default parameters.
In sharpen mode, the training loss I finally achieved is 0.59, and in non-Sharpen mode, the training loss was reduced to 0.38 .
I am curious about what value can the loss of the model converge to in the end during training?

**Comment by rpautrat** [comment]
Hi, for our own training on the wireframe dataset, the final loss value was 0.54. But it is hard and uninformative to compare raw loss values, since there are multiple factors influencing it.

I recommend using the sharpen mode, this will produce sharper distance fields and cleaner lines overall.

**Comment by AIQuantScience** [comment]
Many thanks for your reply。
https://github.com/cvg/DeepLSD/issues/20#issuecomment-1573680070
I noticed that under this question, you mentioned adjusting the input size of the image to 512x512(the default size is 333x333), what other parameters need to be adjusted in train_wireframe.yaml to get the same performance as the model you gave?

**Comment by rpautrat** [comment]
We only used 512x521 images for the outdoor model trained on MegaDepth. The model trained on the wireframe dataset was using the default size of 333x333.

The default parameters of this repo are the ones that led to the released model, so no changes should be necessary.

I would not compare your retrained model to the released one with the validation loss, as this is not a deterministic test (the homographies used in the validation are random and different at each iteration). You should instead consider the evaluation metrics and datasets proposed here: https://github.com/cvg/DeepLSD/tree/main#low-level-line-detection-metrics

**Comment by MHSHAO95** [comment]
hello,I found something puzzling in the VGGUNet code,
When performing feature concat operations, why reverse the Batch dimension of features?
https://github.com/cvg/DeepLSD/blob/main/deeplsd/models/backbones/vgg_unet.py#L92C73-L92C80

**Comment by AIQuantScience** [comment]
Hello,When the two training sets are mixed, will the final model be better than the model trained by a single training set?How much will it increase in terms of metrics?

**Comment by rpautrat** [comment]
Hi, I can't tell since I did not try myself. But my guess is that it will be in between the other two models, both on indoors and outdoor data. So a kind of average between the two.
